### ISSUE TITLE
Align CSS custom properties with Lumo

### ISSUE THREAD

**Issue by warpech** [Starcounter/uniform.css#121]
Align CSS custom property names 1:1 with Lumo (e.g. `--uni-success-text-color` should be our version of `--lumo-success-text-color`).

Use the new property names in place of old property names in stylesheets. We don't care about backward compatibility of CSS custom properties in this particular case.

Provide a translation layer from `--uni-...` to `--lumo-...` so that Vaadin Date Picker will use Uniform configuration (e.g. `--lumo-success-text-color: var(--uni-success-text-color);`)

**Comment by warpech** [comment]
Rolf investigated the use of CSS custom properties in Lumo and Material theme. It looks like Lumo might not be semantic enough, whereas Material is very semantic but seems to be few steps behind in terms of development.

Rolf will clarify with Jouni is Lumo going to be more literal or more semantic in future?

**Comment by rolfsmeds** [comment]
Some further info on Vaadin Lumo and Material themes:

The Material theme is currently in alpha, and will be released as part of Vaadin 11, whenever that will be.

The versioning of the Vaadin platform has changed as of Vaadin 10 so that 11, 12 etc will correspond roughly to minor versions in the old Framework, so 11 does not mean "three years from now" as it would've done for the Framework, but I still have no info on when that will be.

In any case, this does mean that the Material theme will have "full citizenship" in the Vaadin platform, so there's no risk of it suddenly being end-of-lifed.

Lumo might get more semantic css variables in the future, but there are currently no concrete plans on this.

It is also possible that Lumo and Material will be refactored so that they will share a common base theme, and thus a common set of css variables. Again, there is no roadmap for this, nor any info on how semantic these shared variables will be.

TL;DR: Material could very well be a viable alternative to Lumo, but I'm not at all familiar with it so I can't say how mature it is at the moment. This could be investigated of course.

**Comment by alshakero** [comment]
I fail to consildate those two parts:

> Align CSS custom property names 1:1 with Lumo (e.g. `--uni-success-text-color` should be our version of `--lumo-success-text-color`).

> Provide a translation layer from `--uni-...` to `--lumo-...` so that Vaadin Date Picker will use Uniform configuration (e.g. `--lumo-success-text-color: var(--uni-success-text-color);`)

Does the first part I should rename the `--uni-success-text-color` to `--lumo-success-text-color` or just means copy the values while keeping variables name as is. I think the easiest way is to give me an example of the two points.

**Comment by warpech** [comment]
We want to specify a set of primitives that are CSS variables. These primitives are declarations of semantic colors, such as what is the "primary color".

The intended usage for these primitives is to provide the default font and color scheme for Uniform. 

CSS variables are supposed to be documented in Uniform Docs. They can be changed by the solution owner based on the following information: https://github.com/Starcounter/Docs/blob/82c3af002191789b420a52e00c69afe97f0f22ff/docs/topic-guides/blendable-web-apps/uniform-design-system.md#theme-configuration-through-css-custom-properties

These primitives need to affect the look and feel of native HTML (Underwear) as well as Uniform components (which are based on Vaadin).

**Since Vaadin have already created a universal set of primitives (`--lumo`), the simple solution seems to ditch the currently used Uniform and Underwear CSS variables and just make a `--lumo` ripoff with a `--uni` prefix.**

We will need a 1:1 translation layer to Lumo (`--lumo-success-text-color: var(--uni-success-text-color);`) to make it possible to style Uniform components based on `--uni` primitives.
